### PR TITLE
ci: build and deploy with GitHub Actions

### PR DESCRIPTION
**Pull request by lwille**
related to INFRA-1266

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -67,16 +69,14 @@ jobs:
...
-    - name: Publish package to Camunda Nexus (${{ steps.release-detection.outputs.type }})
+      run: mvn -B clean install
+    - name: Publish package (${{ steps.release-detection.outputs.type }})
       run: |-
-        mvn -B -DskipTests=true -Dgpg.passphrase="${GPG_PASSPHRASE}" deploy
-    - name: Publish package to Maven Central (${{ steps.release-detection.outputs.type }})
-      run: |-
-        echo -e "${MVN_CENTRAL_GPG_KEY_SEC}" | gpg -q --allow-secret-key-import --import
-        echo -e "${MVN_CENTRAL_GPG_KEY_PUB}" | gpg -q --allow-secret-key-import --import
+        mvn -B deploy -DskiptTests=true -Dgpg.passphrase="${GPG_PASSPHRASE}"
```
Comment (saig0) [inline .github/workflows/deploy.yml:75]:
```suggestion
        mvn -B deploy -Dgpg.passphrase="${GPG_PASSPHRASE}"
```

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -67,16 +69,14 @@ jobs:
...
+
         sed -i 's@:camunda-nexus:.*$@:central:https://oss.sonatype.org/content/repositories/snapshots/@g' target/nexus-staging/deferred/.index
-        mvn -B -DskipTests=true -s $GITHUB_WORKSPACE/settings.xml -Dgpg.passphrase="${GPG_PASSPHRASE}" org.sonatype.plugins:nexus-staging-maven-plugin:deploy-staged
+        cat target/nexus-staging/deferred/.index
+        mvn -s $GITHUB_WORKSPACE/settings.xml -B org.sonatype.plugins:nexus-staging-maven-plugin:deploy-staged -DskipTests=true -Dgpg.passphrase="${GPG_PASSPHRASE}"
```
Comment (saig0) [inline .github/workflows/deploy.yml:79]:
```suggestion
        mvn -s $GITHUB_WORKSPACE/settings.xml -B org.sonatype.plugins:nexus-staging-maven-plugin:deploy-staged -Dgpg.passphrase="${GPG_PASSPHRASE}"
```

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (lwille) [reply .github/workflows/deploy.yml:75]: we already ran the tests in the previous step with `clean install`, so I thought we can safely skip them here.

Comment (saig0) [reply .github/workflows/deploy.yml:75]: Ah. You're right :sweat_smile:

**Pull request merged**